hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieStreamingSink.scala

```diff
@@ -318,14 +318,18 @@ class HoodieStreamingSink(sqlContext: SQLContext,
 
   private def canSkipBatch(incomingBatchId: Long, operationType: String): Boolean = {
     if (!DELETE_OPERATION_OPT_VAL.equals(operationType)) {
-      val identifier = options.getOrElse(STREAMING_CHECKPOINT_IDENTIFIER.key(), STREAMING_CHECKPOINT_IDENTIFIER.defaultValue())
-      // get the latest checkpoint from the commit metadata to check if the microbatch has already been processed or not
-      val lastCheckpoint = CommitUtils.getValidCheckpointForCurrentWriter(
-        metaClient.get.getActiveTimeline.getWriteTimeline, SINK_CHECKPOINT_KEY, identifier)
-      if (lastCheckpoint.isPresent) {
-        latestCommittedBatchId = lastCheckpoint.get().toLong
+      if(incomingBatchId == 0) {
+         false
```

Comment:
The batch 0 may have already been processed so the last checkpoint still needs to be read.  Correct?

Reply:
Yes, I agree @yihua . Thanks. Closing this PR then without committing. @nsivabalan  Let me know if any concerns.